pina/geometry/union.py

```diff
@@ -22,23 +22,30 @@ def __init__(self, geometries):
             >>> union = GeometryUnion([ellipsoid1, ellipsoid2])
 
         """
-        for geometry in geometries:
-            check_consistency(geometry, Location, type(geometry))
+        for idx, geometry in enumerate(geometries):
```

Comment:
please make a unique private method for this. Also the `super().__init__()` should be call right after the docs end as first command.